### PR TITLE
HP-84: Support for giving rewards in multiple tokens

### PR DESCRIPTION
**Pull request by ankurdubey521**
- Implement `migrateNftsToV2` in LiquidityFarming to allow users to migrate their NFTs to V2 contract.
- Implement LiquidityFarmingV2 to introduce support for giving rewards in multiple tokens simultaneously.

### CODE REVIEW

File: contracts/hyphen/LiquidityFarming.sol [hunk 1]
```diff
@@ -84,6 +85,7 @@ contract HyphenLiquidityFarming is
     event LogRewardPoolInitialized(address _baseToken, address _rewardToken, uint256 _rewardPerSecond);
     event LogNativeReceived(address indexed sender, uint256 value);
     event LiquidityProviderUpdated(address indexed liquidityProviders);
+    event LogNftMigrated(address indexed owner, uint256 indexed nftId, IHyphenLiquidityFarmingV2 farmingContract);
```
Comment (tomarsachin2271) [inline contracts/hyphen/LiquidityFarming.sol:88]: In the logs, IHyphenLiquidityFarmingV2 address will go or the whole interface?? I think we just need the address to be logged.

File: contracts/hyphen/LiquidityFarming.sol [hunk 2]
```diff
@@ -381,6 +383,40 @@ contract HyphenLiquidityFarming is
...
+
+        for (uint256 i = 0; i < length; ) {
+            uint256 nftId = _nftIds[i];
+            uint256 index = getStakedNftIndex(_msgSender(), nftId);
+
+            require(nftIdsStaked[msgSender][index] == nftId, "ERR__NOT_OWNER");
+            require(nftInfo[nftId].staker == msgSender, "ERR__NOT_OWNER");
+
+            nftIdsStaked[msgSender][index] = nftIdsStaked[msgSender][nftIdsStaked[msgSender].length - 1];
+            nftIdsStaked[msgSender].pop();
```
Comment (tomarsachin2271) [inline contracts/hyphen/LiquidityFarming.sol:398]: What is being done here?

File: contracts/hyphen/LiquidityFarming.sol (hunk 1 above)
Comment (ankurdubey521) [reply contracts/hyphen/LiquidityFarming.sol:88]: Just the address would be logged, in these cases solidity automatically down casts interfaces to addresses.

File: contracts/hyphen/LiquidityFarming.sol (hunk 2 above)
Comment (ankurdubey521) [reply contracts/hyphen/LiquidityFarming.sol:398]: We are deleting the element at index `index` by copying the last element in the array to the position `index` and then reducing the length of the array by 1

Comment (tomarsachin2271) [reply contracts/hyphen/LiquidityFarming.sol:398]: Got it.. i thought the loop is also applied on same array.

**Review by tomarsachin2271 — APPROVED** [review]

**Pull request merged**